platform/posix/transport/utest/reconnect_utest.c

```diff
@@ -0,0 +1,128 @@
+
+/**
+ * @brief A helper method to make assertions on the #TransportReconnectParams_t
+ * object upon calling #Transport_ReconnectParamsReset.
+ */
+static void verifyReconnectParamsAfterReset( void )
+{
+    TEST_ASSERT_EQUAL( 0, reconnectParams.attemptsDone );
+    TEST_ASSERT_GREATER_OR_EQUAL_INT32( INITIAL_RECONNECT_BACKOFF_SECONDS,
+                                        reconnectParams.nextJitterMax );
+    TEST_ASSERT_LESS_THAN_INT32( ( INITIAL_RECONNECT_BACKOFF_SECONDS +
+                                   MAX_JITTER_VALUE_SECONDS ),
+                                 reconnectParams.nextJitterMax );
+}
+
+/**
+ * @brief Test that #Transport_ReconnectParamsReset initializes the #nextJitterMax
+ * to be equal to the some configurable backoff plus jitter.
+ */
+void test_Transport_ReconnectParamsReset_Sets_Jitter_Correctly( void )
+{
+    Transport_ReconnectParamsReset( &reconnectParams );
+    verifyReconnectParamsAfterReset();
+}
+
+/**
+ * @brief Test that #Transport_ReconnectBackoffAndSleep is able to double the
+ * max value of the next jitter and return false when all attempts are exhausted.
+ *
+ * @note #Transport_ReconnectParamsReset is expected to be called before
+ * #Transport_ReconnectBackoffAndSleep.
+ */
+void test_Transport_ReconnectBackoffAndSleep_Succeeds( void )
+{
+    bool retriesArePending;
+    int32_t expectedNextJitterMax = INITIAL_RECONNECT_BACKOFF_SECONDS;
+
+    Transport_ReconnectParamsReset( &reconnectParams );
+    verifyReconnectParamsAfterReset();
+
+    expectedNextJitterMax = reconnectParams.nextJitterMax;
+
+    while( reconnectParams.attemptsDone < MAX_RECONNECT_ATTEMPTS )
+    {
+        if( reconnectParams.nextJitterMax < ( MAX_RECONNECT_BACKOFF_SECONDS / 2U ) )
+        {
+            expectedNextJitterMax += expectedNextJitterMax;
+        }
+        else
+        {
+            expectedNextJitterMax = MAX_RECONNECT_BACKOFF_SECONDS;
+        }
+
+        /* Mock sleep, so the test can run faster. */
+        sleep_ExpectAnyArgsAndReturn( 0 );
+        retriesArePending = Transport_ReconnectBackoffAndSleep( &reconnectParams );
+        TEST_ASSERT_TRUE( retriesArePending );
+        TEST_ASSERT_EQUAL( expectedNextJitterMax, reconnectParams.nextJitterMax );
+    }
+
+    /* The next call to the function under test should now return that all
+     * attempts are now exhausted. */
+    retriesArePending = Transport_ReconnectBackoffAndSleep( &reconnectParams );
+    TEST_ASSERT_FALSE( retriesArePending );
+
+    /* #Transport_ReconnectParamsReset is expected to be called once all attempts
+     * are exhausted. */
+    verifyReconnectParamsAfterReset();
+}
+
+/**
+ * @brief Test that the value of the next max jitter is capped at some max threshold.
+ */
+void test_Transport_ReconnectBackoffAndSleep_Jitter_Has_Max( void )
+{
+    bool retriesArePending;
+
+    /* Initialize to 0 attempts, so the max value of next jitter will increase. */
+    reconnectParams.attemptsDone = 0;
```

Comment:
Also, add  a test when the initialized `nextJitterMax` value is `MAX_RECONNECT_BACKOFF_SECONDS - 1`